Assignment 3.Rmd

```diff
@@ -127,29 +136,189 @@ Now we can plot our line plot using the ggplot command, "ggplot()".
+      DF2[[i]][j]<-0
+      b<-mean(as.numeric(DF2[[i]]))
+      DF2[[i]][j]<-b
+    }
+  }
+}
+
+#DF2%>%mutate_all(as.character)%>%mutate_all(as.numeric)
+
+DF2<-data.frame(DF2)
+indx <- sapply(DF2, is.factor)
+DF2[indx] <- lapply(DF2[indx], function(x) as.numeric(as.character(x))) #convert to numeric
+
+colnames(DF2)<-c("month live in NYC","# siblings","# sport each week","distance to TC","#friend on phone","# movies","#pet own","#people met this year","#cook","#class taking","#states traveled")
+
+#numberic??????? easy way to convert whole graph to numeric
+#Scale the data so that no variable has undue influence
+
+DF2 <- scale(DF2)
+ 
+```
+
+
+
+```{r}
+#install.packages("ggplot2")
+library(ggplot2)
+fit <- kmeans(DF2, 4) 
+
+#We have created an object called "fit" that contains all the details of our clustering including which observations belong to each cluster.
+
+#We can access the list of clusters by typing "fit$cluster", the top row corresponds to the original order the rows were in. Notice we have deleted some rows.
+
+cluster2<-fit$cluster
+
+#We can also attach these clusters to tje original dataframe by using the "data.frame" command to create a new data frame called K4.
+
+DF3 <- data.frame(DF2,cluster2)
+
+#Have a look at the DF3 dataframe. Lets change the names of the variables to make it more convenient with the names() command.
+
+names(DF3) <- c("1", "2", "3", "4", "5","6","7","8","9","10","11","cluster2")
+row.names(DF3)<-DF1$"1"
+datacluster2<-dplyr::select(DF3,"cluster2") #only want name & cluster easy to combine
+name1<-DF1$"1"
+datacluster2<-data.frame(name1,datacluster2)
+#convert to upper case alphabet
+datacluster1[,1]<-toupper(datacluster1[,1])
+datacluster2[,1]<-toupper(datacluster2[,1])
+#inner join the table by name (delete the name that do not match)
+CC<-dplyr::inner_join(datacluster1,datacluster2,by=c("name"="name1"))
+graph<-table(CC$cluster1,CC$cluster2)
+CC$cluster2<-as.numeric(CC$cluster2) #convert to numeric
+CC$studentID<-c(1:33)
+#cluster 1 scatterplot+cluster2 scatterplot show the overlap
+#g1<-ggplot()+geom_point(data=CC,aes(x=CC$studentID,y=CC$cluster1,pch=16),size=5)+
+  #geom_point(data=CC,aes(x=CC$studentID,y=CC$cluster2,pch=6),size=5)+
+ # scale_shape_identity()
+  #scale_fill_discrete(breaks=c("red"))
+# in order to make the plot depend on cluster, I will gather data in order to make the plot categorize by cluster
+CC1<-CC%>%select(-name)%>%gather(cluster_type, cluster, 1:2, -studentID)
+#plot the cluster
+ggplot(data=CC1, aes(x=studentID, y=cluster, color=cluster_type, shape=cluster_type))+
+  geom_point(size=5)
+
+#try another way!
+#install.packages("vcd")
+library(vcd)
+mosaicplot(graph, color=c(2,3,4,6))
+#+scale_shape_identity()
+#c() stands for concatonate and it creates a vector of anything, in this case a vector of names.
+```
+```{r}
+#compare two cluster tried several methods
+#adj.rand.index(cluster1,cluster2)
+#HeatmapPlot(cluster1,cluster2)
+#doesn't work?
+
+```
```

Comment:
Overall, great job on the coding part of this assignment but you may want to work on the analysis aspect of it. Nonetheless, keep up the good work!